grr/server/grr_response_server/gui/api_plugins/metadata.py

```diff
@@ -952,3 +1003,116 @@ def _GetMapFieldKeyValueTypes(
 def _IsMapField(field_descriptor: FieldDescriptor) -> bool:
   """Checks that a `FieldDescriptor` is of a map type."""
   return _GetMapFieldKeyValueTypes(field_descriptor) is not None
+
+
+class ComponentTrieNode:
+  def __init__(self, component, parent_path):
+    self.component = component
+    if parent_path:
+      self.path = f"{parent_path}/{component}"
+    else:
+      self.path = component
+    self.is_path_arg = component.startswith("<") and component.endswith(">")
+    self.is_route_end = False
+    self.children = dict()
+
+
+def _CreateTrie(routes: List[List[str]]) -> ComponentTrieNode:
```

Comment:
I would make it a pseudo-constructor (a `@classmethod` that returns a class instance) called `FromRoutes`.

Reply:
Made the change. Thank you! A note about the `curr_node` local variable: I annotated it as a `ComponentTrieNode` so I can keep `ComponentTrieNode.children` as a dictionary of `ComponentTrieNode`. This way, if we ever subclass ComponentTrieNode, the `FromRoutes` class method would still work, while allowing tries with nodes of different subclasses of `ComponentTrieNode` to exist. I don't know how useful/probable this is to happen, but I believe this annotation makes sense. Please let me know if you think differently.